core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushJoinIntoTableScan.java

```diff
@@ -114,6 +115,19 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
             return Result.empty();
         }
 
+        if (getJoinPushdownMode(context.getSession()) == JoinPushdownMode.AUTOMATIC) {
+            PlanNodeStatsEstimate joinStats = context.getStatsProvider().getStats(joinNode);
+            PlanNodeStatsEstimate leftStats = context.getStatsProvider().getStats(joinNode.getLeft());
+            PlanNodeStatsEstimate rightStats = context.getStatsProvider().getStats(joinNode.getRight());
+            if (joinStats.isOutputRowCountUnknown() || leftStats.isOutputRowCountUnknown() || rightStats.isOutputRowCountUnknown()) {
+                return Result.empty();
+            }
+
+            if (joinStats.getOutputRowCount() > leftStats.getOutputRowCount() + rightStats.getOutputRowCount()) {
```

Comment:
While this is not a rocket science, it'd be nice to add some comment, eg why we're choosing `+` over `max`.
from my perspective it was some 'random thought from findepi' (and i don't feel strongly), but still let's safe future readers suffering and try to word some explanation.

Reply:
I added some reasoning. Not sure if helpful